src/sagemaker/jumpstart/model.py

```diff
@@ -786,6 +803,86 @@ def register_deploy_wrapper(*args, **kwargs):
 
         return model_package
 
+    @property
+    def benchmark_metrics(self) -> pd.DataFrame:
+        """Pandas DataFrame object of Benchmark Metrics for deployment configs"""
+        if self._benchmark_metrics:
+            return self._benchmark_metrics
+
+        data = extract_metrics_from_deployment_configs(
+            deployment_configs=self.list_deployment_configs(),
+            config_name=self.config_name,
+        )
+
+        self._benchmark_metrics = pd.DataFrame(data)
+        return self._benchmark_metrics
+
+    def display_benchmark_metrics(self):
+        """Display Benchmark Metrics for deployment configs."""
+        print(self.benchmark_metrics.to_markdown())
+
+    def list_deployment_configs(self) -> List[Dict[str, Any]]:
+        """List deployment configs for ``This`` model in the current region.
+        Returns:
+            A list of deployment configs (List[Dict[str, Any]]).
+        """
+        if self._metadata_configs is None:
+            self._metadata_configs = get_jumpstart_configs(
+                region=self.region,
+                model_id=self.model_id,
+                model_version=self.model_version,
+                sagemaker_session=self.sagemaker_session,
+            )
+
+            self._deployment_configs = [
+                self._convert_to_deployment_config_metadata(config_name, config)
+                for config_name, config in self._metadata_configs.items()
+            ]
```

Comment:
Can we generate all these class fields at class initialization time? I feel like that'll make the logic simpler